source/dub/commandline.d

```diff
@@ -77,6 +78,314 @@ CommandGroup[] getCommands()
 	];
 }
 
+/** Extract the command name from the argument list
+
+	Params:
+		args = a list of string arguments that will be processed
+
+	Returns:
+		A structure with two members. `value` is the command name
+		`remaining` is a list of unprocessed arguments
+*/
+auto extractCommandNameArgument(string[] args)
+{
+	struct Result {
+		string value;
+		string[] remaining;
+	}
+
+	if (args.length >= 1 && !args[0].startsWith("-")) {
+		return Result(args[0], args[1 .. $]);
+	}
+
+	return Result(null, args);
+}
+
+/// It returns an empty string on when there are no args
+unittest {
+	assert(extractCommandNameArgument([]).value == "");
+	assert(extractCommandNameArgument([]).remaining == []);
+}
+
+/// It returns the first argument when it does not start with `-`
+unittest {
```

Comment:
These unittests could be merged because they test the same function.

Reply:
Why do you think that would be better? I see an advantage of having separate  unit tests, because in case of a change, you can easily see which assets failed. When you have everything in one unit test block, you see only the first failing assert.

Reply:
I agree that more separate unittests are in general better, but here you are extracting the same `["test"]` input parameter so what MoonlightSentinel is trying to say is that the .value and .remaining test for the same input should be in the same unittest block I think.

Reply:
ok. done.